examples/views/new_components.py

```diff
@@ -0,0 +1,81 @@
+from io import BytesIO
+
+from discord import (
+    ApplicationContext,
+    Bot,
+    ButtonStyle,
+    Color,
+    File,
+    Interaction,
+    SeparatorSpacingSize,
+    User,
+)
+from discord.ui import (
+    Button,
+    Container,
+    MediaGallery,
+    Section,
+    Select,
+    Separator,
+    TextDisplay,
+    Thumbnail,
+    View,
+    button,
+)
+
+
+class MyView(View):
+    def __init__(self, user: User):
+        super().__init__(timeout=30)
+        text1 = TextDisplay("### This is a sample `TextDisplay` in a `Section`.")
+        text2 = TextDisplay(
+            "This section is contained in a `Container`.\nTo the right, you can see a `Thumbnail`."
+        )
+        thumbnail = Thumbnail(user.display_avatar.url)
+
+        section = Section(text1, text2, accessory=thumbnail)
+        section.add_text("-# Small text")
+
+        container = Container(
+            section,
+            TextDisplay("Another `TextDisplay` separate from the `Section`."),
+            color=Color.blue(),
+        )
+        container.add_separator(divider=True, spacing=SeparatorSpacingSize.large)
+        container.add_item(Separator())
+        container.add_file("attachment://sample.png")
+        container.add_text("Above is two `Separator`s followed by a `File`.")
+
+        gallery = MediaGallery()
+        gallery.add_item(user.default_avatar.url)
+        gallery.add_item(user.avatar.url)
+
+        self.add_item(container)
+        self.add_item(gallery)
+        self.add_item(
+            TextDisplay("Above is a `MediaGallery` containing two `MediaGalleryItem`s.")
+        )
+
+    @button(label="Delete Message", style=ButtonStyle.red, id=200)
+    async def delete_button(self, button: Button, interaction: Interaction):
+        await interaction.response.defer(invisible=True)
+        await interaction.delete_original_response()

```

Comment:
```suggestion
        await interaction.message.delete()

```

Reply:
I think officially `delete_original_response` is recommended because defer is a response?

Reply:
`delete_original_response` deletes whatever response you sent, it looks like what you want is to delete the message the button is on

Reply:
I think having an edit button would be more useful in this scenario anyways, then the `id` functionality can be shown. That change can be made in a future PR though.